pom.xml

```diff
@@ -783,6 +783,7 @@
             <showWarnings>true</showWarnings>
             <compilerArgs>
               <arg>-Xlint:all</arg>
+              <arg>-Xlint:-this-escape</arg>
```

Comment:
I don't think this is a good one to ignore. This one catches problems of improperly constructed classes calling methods from uninitialized subclasses while the parent class is in the process of being constructed. These shouldn't be ignored, they should be fixed.

Reply:
Could add suppression in the code of the places it happens now and open follow on issues, but not suppress it everywhere.

Reply:
I'd rather just fix them first. It seems like those could be legitimate bugs. There's no rush to get this in. I'd like to understand the scope of the problem first, though. If you'll indulge me and hold off, I could spend some time trying to track these down and knock them out next week.